### PR TITLE
Improve Table Aggregation UI

### LINKED ISSUE #4629: Table Aggregation dialog UI wastes space
**Describe the bug**
Essentially the current Table Aggregation dialog is a single long list of fields. It looks auto-generated and does not use the space efficiently. It also makes the preview section seem like it is part of the items you can modify above, which further confuses the user. We could use a grid to collapse it a bit and use the space better while making it more intuitive for the user.

image

### PR DESCRIPTION
**Pull request by grantfitzsimmons**
Fixes #4629

> image
> 
> I added an explanation and highlighted "Preview" to help users segment and understand that section better. Feedback is welcome on this as well during the review process 😊 

Left is `issue-4629`, right is `edge`

https://github.com/specify/specify7/assets/37256050/fb300615-daf3-446f-bd12-bac2d3b41598

(Ignore the localization, that is not relevant for this PR)

### Checklist

- [X] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [X] Add relevant issue to release milestone

### Testing instructions
- [ ] Verify that all of the functionality in `edge` is preserved when customizing table aggregation
- [ ] Verify that the new layout is more intuitive for the user than the previous layout
- [ ] Configure several table aggregations and test them in the query builder and preview (please mention which ones you created and include screenshots)
- [ ] Provide feedback on interface if necessary

### CODE REVIEW

**Comment by grantfitzsimmons** [comment]
image

I added an explanation and highlighted "Preview" to help users segment and understand that section better. Feedback is welcome on this as well during the review process 😊

**Review by bronwyncombs — APPROVED** [review]
Condensed UI looks good and functions as expected. Description for search is logical and helpful!

Screenshot 2024-03-11 at 9 50 13 AM

File: specifyweb/frontend/js_src/lib/components/Formatters/Aggregator.tsx [hunk 1]
```diff
@@ -66,36 +97,9 @@ export function AggregatorElement({
...
-          onValueChange={(limit): void =>
-            setAggregator({
-              ...aggregator,
-              limit,
-            })
-          }
-        />
-      </Label.Block>
-      <AggregatorPreview aggregator={aggregator} />
+      <div>
```
Comment (CarolineDenis) [inline specifyweb/frontend/js_src/lib/components/Formatters/Aggregator.tsx:100]: this div should not be needed

Comment (grantfitzsimmons) [reply specifyweb/frontend/js_src/lib/components/Formatters/Aggregator.tsx]: thank you! 😄

**Review by CarolineDenis — APPROVED** [review]

**Review by maxpatiiuk — APPROVED** [review]

**Pull request merged**